components/centraldashboard/public/components/main-page.js

```diff
@@ -133,21 +151,24 @@ export class MainPage extends PolymerElement {
             this.sidebarItemIndex = 0;
             this.page = 'dashboard';
         }
+        const isIframe = newPage == '_';
```

Comment:
The following logic works for all blocks not just the iframe case, since I check if state has changed. So even iframe -> dashboard, will trigger the code block

Reply:
Understood. It felt a little bit odd to me since some of the logic was dependent on the page being '_' and it was found in two places. I also think it's generally easier to read structured conditionals as opposed to the more bash-like style using && to chain commands.

```suggestion
_routePageChanged(newPage) {
    let changedToIframe = false;
    switch (newPage) {
    case 'activity':
        this._setInIframe(false);
        this.sidebarItemIndex = 0;
        this.page = 'activity';
        break;
    case '_': // iframe case
        changedToIframe = !this.inIframe;
        this._setInIframe(true);
        this._setIframeFromRoute(this.subRouteData.path);
        break;
    default:
        this._setInIframe(false);
        this.sidebarItemIndex = 0;
        this.page = 'dashboard';
    }
    if (changedToIframe) {
        this.$.MainDrawer.close();
    }
}
```

Reply:
Resolved in a slightly different approach.